Assets/Scripts/HealthBar.cs

```diff
@@ -0,0 +1,39 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+/*
+ * UI player health bar
+ * Displays player health remaining using a heart tally
+ */
+public class HealthBar : MonoBehaviour
+{
+    public Robo player;
+    public Heart[] hearts;
+    public int max_hearts;
+    public Sprite filled_heart;
+    public Sprite empty_heart;
+    
+    void Start() {
+        max_hearts = Mathf.Clamp(player.max_health, 0, hearts.Length);
```

Comment:
Yeah, it is just out of a lot of caution, far more than displayed elsewhere, mainly because it's not my caution. While I was researching other implementations of similar health systems, I found one that performed that check, and just decided to include it because it didn't seem to cause too great of computational overhead since it only has to run once.

I don't think it's *really* necessary. If you think it's extraneous, I'm good with getting rid of it, as I don't plan on setting a player's max_health to negative values or values greater than the length of the health bar.

Reply:
From verbal discussion, this was deemed necessary and retained (by which I mean to say, rewritten because I deleted it.)